stellarphot/tests/test_core.py

```diff
@@ -25,74 +25,172 @@
             pixel_scale=pixel_scale,
+            max_data_value=max_val,
         )
-    with pytest.raises(ValidationError, match="dark_current"):
-        c = Camera(
+
+
+def test_camera_incompatible_gain_units():
+    data_unit = u.adu
+    gain = 2.0 * u.count / u.adu
+    read_noise = 10 * u.electron
+    dark_current = 0.01 * u.electron / u.second
+    pixel_scale = 0.563 * u.arcsec / u.pix
+    max_val = 50000 * u.adu
+
+    # Make sure that an incompatible gain raises an error
+    with pytest.raises(ValidationError, match="Gain units.*not compatible"):
+        Camera(
+            data_unit=u.adu,
             gain=gain,
             read_noise=read_noise,
-            dark_current=dark_current.value,
+            dark_current=dark_current,
             pixel_scale=pixel_scale,
+            max_data_value=max_val,
         )
-    with pytest.raises(ValidationError, match="pixel_scale"):
-        c = Camera(
+
+
+def test_camera_incompatible_max_val_units():
+    data_unit = u.adu
+    gain = 2.0 * u.electron / u.adu
+    read_noise = 10 * u.electron
+    dark_current = 0.01 * u.electron / u.second
+    pixel_scale = 0.563 * u.arcsec / u.pix
+    max_val = 50000 * u.count
+
+    # Make sure that an incompatible gain raises an error
+    with pytest.raises(
+        ValidationError, match="Maximum data value units.*not consistent"
+    ):
+        Camera(
+            data_unit=u.adu,
             gain=gain,
             read_noise=read_noise,
             dark_current=dark_current,
-            pixel_scale=pixel_scale.value,
+            pixel_scale=pixel_scale,
+            max_data_value=max_val,
         )
 
 
+def test_camera_copy():
+    # Make sure copy actually copies everything
+    gain = 2.0 * u.electron / u.adu
+    read_noise = 10 * u.electron
+    dark_current = 0.01 * u.electron / u.second
+    pixel_scale = 0.563 * u.arcsec / u.pix
+    c = Camera(
+        data_unit=u.adu,
+        gain=gain,
+        read_noise=read_noise,
+        dark_current=dark_current,
+        pixel_scale=pixel_scale,
+        max_data_value=65535 * u.adu,
+    )
+    c2 = c.copy()
+    assert c2 == c
+
+
 def test_camera_altunitscheck():
     # Check to see that 'count' is allowed instead of 'electron'
+    data_unit = u.adu
     gain = 2.0 * u.count / u.adu
     read_noise = 10 * u.count
     dark_current = 0.01 * u.count / u.second
     pixel_scale = 0.563 * u.arcsec / u.pix
+    max_val = 50000 * u.adu
+
     c = Camera(
+        data_unit=data_unit,
         gain=gain,
         read_noise=read_noise,
         dark_current=dark_current,
         pixel_scale=pixel_scale,
+        max_data_value=max_val,
     )
+    assert c.data_unit == data_unit
     assert c.gain == gain
     assert c.dark_current == dark_current
     assert c.read_noise == read_noise
     assert c.pixel_scale == pixel_scale
+    assert c.max_data_value == max_val
+
+
+def test_camera_schema():
+    # Check that we can generate a schema for a Camera and that it
```

Comment:
I assume this schema method is a feature of pydantic...

Reply:
Yeah; it is used by ipyautoui when generating the UI.